graphstorm-processing/tests/test_dist_heterogenous_loader.py

```diff
@@ -0,0 +1,662 @@
+            assert (
+                metadata["edge_data"][rev_type]["train_mask"]
+                == metadata["edge_data"][edge_type]["train_mask"]
+            )
+            assert (
+                metadata["edge_data"][rev_type]["val_mask"]
+                == metadata["edge_data"][edge_type]["val_mask"]
+            )
+
+
+def test_write_edge_structure_no_reverse_edges(
+    data_configs_with_label,
+    dghl_loader_no_reverse_edges: DistHeterogeneousGraphLoader,
+    user_rated_movie_df: DataFrame,
+):
+    """Writing edge structure when no reverse edges are requested"""
+    edge_configs = data_configs_with_label["edges"]
+
+    # We need these two for the side-effects of creating the mappings
+    missing_node_types = dghl_loader_no_reverse_edges._get_missing_node_types(edge_configs, [])
+    dghl_loader_no_reverse_edges.create_node_id_maps_from_edges(edge_configs, missing_node_types)
+
+    edge_dict: Dict[str, Dict] = {
+        "data": {"format": "csv", "files": ["edges/user-rated-movie.csv"], "separator": ","},
+        "source": {"column": "~from", "type": "user"},
+        "relation": {"type": "rated"},
+        "dest": {"column": "~to", "type": "movie"},
+    }
+
+    edge_config = EdgeConfig(edge_dict, edge_dict["data"])
+    _, _, rev_edge_path_list = dghl_loader_no_reverse_edges.write_edge_structure(
+        user_rated_movie_df, edge_config
+    )
+
+    assert len(rev_edge_path_list) == 0
+
+
+def test_create_all_mapppings_from_edges(
+    data_configs_with_label, dghl_loader: DistHeterogeneousGraphLoader
+):
+    """Test creating all node mappings only from edge files"""
+    edge_configs = data_configs_with_label["edges"]
+
+    missing_node_types = dghl_loader._get_missing_node_types(edge_configs, [])
+    dghl_loader.create_node_id_maps_from_edges(edge_configs, missing_node_types)
+
+    assert len(dghl_loader.node_mapping_paths) == 4
+    # TODO: Check mapping contents?
+
+
+def test_create_some_mapppings_from_edges(
+    data_configs_with_label, dghl_loader: DistHeterogeneousGraphLoader
+):
+    """Test creating only some node mappings from edge files"""
+    edge_configs = data_configs_with_label["edges"]
+
+    dghl_loader.create_node_id_maps_from_edges(edge_configs, missing_node_types={"director"})
+
+    assert len(dghl_loader.node_mapping_paths) == 1
+
+
+def test_extend_mapping_from_edges(spark: SparkSession, dghl_loader: DistHeterogeneousGraphLoader):
+    """Test extending an existing node mapping from a new edge source"""
+    existing_mapping_data = [
+        ("node_id_1", 0),
+        ("node_id_2", 1),
+        ("node_id_3", 2),
+    ]
+    edges_data = [
+        ("node_id_1", "dst_id_1", "edge_feature_val1"),
+        ("node_id_4", "dst_id_2", "edge_feature_val2"),
+        ("node_id_3", "dst_id_3", "edge_feature_val3"),
+    ]
+
+    edge_columns = ["src", "dst", "feature"]
+    edges_df = spark.createDataFrame(edges_data, schema=edge_columns)
```

Comment:
Do we need to initialize a spark env to run the unitest?

Reply:
Yes, we initialize a local Spark env, I can add the `conftest.py` that does this in this PR if you'd like. My plan was to add it in a follow-up PR.